### ISSUE TITLE
AssertIncludeCollection ordering of elements issue

### ISSUE THREAD

**Issue by roji** [dotnet/efcore#22293]
When syncing EFCore.PG to the latest rc1 daily, all split tests under TPTRelationshipsQueryNpgsqlTest are failing with incorrect results.

Looking at test Include_collection_with_inheritance_split, the 2nd query for SQL Server is:

```sql
SELECT [b0].[BaseInheritanceRelationshipEntityId], [b0].[Id], [b0].[Name], [b].[Id]
FROM [BaseEntities] AS [b]
INNER JOIN [BaseEntities_OwnedCollectionOnBase] AS [b0] ON [b].[Id] = [b0].[BaseInheritanceRelationshipEntityId]
ORDER BY [b].[Id]
```

For SQL Server, this yields:

| BaseInheritanceRelationshipEntityId | Id | Name | Id |
| :--- | :--- | :--- | :--- |
| 1 | 1 | OCOB11 | 1 |
| 1 | 2 | OCOB12 | 1 |
| 2 | 3 | OCOB21 | 2 |
| 4 | 4 | OCOB41 | 4 |
| 4 | 5 | OCOB42 | 4 |
| 5 | 6 | OCOB51 | 5 |

But PostgreSQL yields:

| BaseInheritanceRelationshipEntityId | Id | Name | Id |
| :--- | :--- | :--- | :--- |
| 1 | 1 | OCOB11 | 1 |
| 1 | 2 | OCOB12 | 1 |
| 2 | 3 | OCOB21 | 2 |
| 4 | 5 | OCOB42 | 4 |
| 4 | 4 | OCOB41 | 4 |
| 5 | 6 | OCOB51 | 5 |

Since the query orders only by [b].[Id] and there are duplicates, row ordering seems non-deterministic...

/cc @smitpatel @AndriySvyryd

**Comment by smitpatel** [comment]
@maumar - Can you investigate why the ordering on the collection is not happening in client side when asserting results?

**Comment by smitpatel** [comment]
Fixed via c57c5cea768276139a185ea75a6a0af2e4bcd015
